template.distillery/PROJECT_NAME_otdemo.eliom

```diff
@@ -58,78 +69,264 @@ end
 
 let%client (carousel_update, carousel_change) = React.E.create ()
 
+let%server service = Eliom_service.create
+  ~id:(Eliom_service.Path ["otdemo-carousel"])
+  ~meth:(Eliom_service.Get Eliom_parameter.unit)
+  ()
+
 [%%shared
 module CarouselPage : DemoPage = struct
+
   let name = "Carousel"
-  let service =
-    Eliom_service.create
-      ~id:(Eliom_service.Path ["otdemo-carousel"])
-      ~meth:(Eliom_service.Get Eliom_parameter.unit)
-      ()
+
+  let service = ~%service
+
   let page () =
-    let make_page content = div ~a:[a_class ["otdemo-carousel-page"]] [pcdata content] in
+    let make_page content =
+      div ~a:[a_class ["otdemo-carousel-page"]] [pcdata content]
+    in
     let carousel_pages = ["1"; "2"; "3"] in
-    let carousel, pos, size, _ = Ot_carousel.make
-      ~a:[a_class ["otdemo-carousel"]]
-      ~update:[%client carousel_update]
-      (List.map make_page carousel_pages)
+    let ribbon_content = List.map (fun p -> [pcdata p]) carousel_pages in
+    let carousel_content = List.map make_page carousel_pages in
+    let bullets_content =
+      List.map (fun n -> [div [p [pcdata n]]]) carousel_pages
     in
-    let prev = Ot_carousel.previous
-      ~a:[a_class ["button"]]
-      ~change:[%client carousel_change]
-      ~pos
-      [pcdata "←"]
+    let carousel,
+      Some ribbon,
+      Some prev,
+      Some next,
+      Some bullets
```

Comment:
Can you make this pattern matching exhaustive? I think we just need a `failwith` with a reasonable message for any case than `Some _, Some _, Some _, Some _`.